src/System.ObjectModel/tests/ObservableCollection/ObservableCollection_MethodsTest.cs

```diff
@@ -454,6 +455,280 @@ public static void GetEnumeratorTest()
+            collection.CollectionChanged += (o, e) => actualDataAdded = e.NewItems.Cast<int>().ToArray();
+
+            collection.AddRange(dataToAdd);
+
+            Assert.NotNull(collection);
+            Assert.Equal(dataToAdd, actualDataAdded);
+        }
+
+        [Fact]
+        public static void InsertRange_NotifyCollectionChanged_EventArgs_Test()
+        {
+            int[] dataToAdd = new int[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+            int[] actualDataAdded = new int[0];
+            ObservableCollection<int> collection = new ObservableCollection<int>();
+            collection.CollectionChanged += (o, e) => actualDataAdded = e.NewItems.Cast<int>().ToArray();
+
+            collection.InsertRange(0, dataToAdd);
+
+            Assert.NotNull(collection);
+            Assert.Equal(dataToAdd, actualDataAdded);
+        }
+
+        [Fact]
+        public static void InsertRange_NotifyCollectionChanged_EventArgs_Middle_Test()
+        {
+            int[] dataToAdd = new int[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+            int[] actualDataAdded = new int[0];
+            ObservableCollection<int> collection = new ObservableCollection<int>();
+            for (int i = 0; i < 4; i++)
+            {
+                collection.Add(i);
+            }
+
+            collection.CollectionChanged += (o, e) => actualDataAdded = e.NewItems.Cast<int>().ToArray();
+            collection.InsertRange(2, dataToAdd);
+
+            Assert.NotNull(collection);
+            Assert.Equal(dataToAdd, actualDataAdded);
+        }
+
+        [Fact]
+        public static void InsertRange_NotifyCollectionChanged_EventArgs_End_Test()
+        {
+            int[] dataToAdd = new int[] { 1, 2, 3, 4, 5, 6, 7, 8 };
+            int[] actualDataAdded = new int[0];
+            ObservableCollection<int> collection = new ObservableCollection<int>();
+            for (int i = 0; i < 4; i++)
+            {
+                collection.Add(i);
+            }
+
+            collection.CollectionChanged += (o, e) => actualDataAdded = e.NewItems.Cast<int>().ToArray();
+            collection.InsertRange(4, dataToAdd);
+
+            Assert.NotNull(collection);
+            Assert.Equal(dataToAdd, actualDataAdded);
+        }
+
+        [Fact]
+        public static void RemoveRange_NotifyCollectionChanged_FirstTwo_Test()
+        {
+            int[] initialData = new int[] { 10, 11, 12, 13 };
+            int itemsToRemove = 2;
+            int eventCounter = 0;
+            ObservableCollection<int> collection = new ObservableCollection<int>(initialData);
+            collection.CollectionChanged += (o, e) => eventCounter++;
+
+            collection.RemoveRange(0, itemsToRemove);
+
+            Assert.NotNull(collection);
+            Assert.Equal(initialData.Length - itemsToRemove, collection.Count);
+            Assert.Equal(1, eventCounter);
+            Assert.Equal(initialData.AsSpan(2, 2).ToArray(), collection.ToArray());
```

Comment:
You shouldn't need the second arg in `AsSpan` right?

Reply:
Correct, the 2nd parameter is not needed here. I will remove it in my update